Services/Simulations.cs

```diff
@@ -210,23 +240,27 @@ public async Task<Models.Simulation> UpsertAsync(Models.Simulation simulation)
                 simulation.Modified = simulation.Created;
             }
 
-            // Note: forcing the ID because only one simulation can be created
-            simulation.Id = SIMULATION_ID;
-
             // TODO if write to storage adapter fails, the iothub connection string 
             //      will still be stored to disk. Storing the encrypted string using
             //      storage adapter would address this
             //      https://github.com/Azure/device-simulation-dotnet/issues/129
-            simulation.IotHubConnectionString = await this.connectionStringManager.RedactAndStoreAsync(simulation.IotHubConnectionString);
+            var iotHubConnectionStrings = new List<string>(simulation.IotHubConnectionStrings);
+            simulation.IotHubConnectionStrings.Clear();
+            foreach (var iotHubConnectionString in iotHubConnectionStrings)
+            {
+                var connString = await this.connectionStringManager.RedactAndStoreAsync(iotHubConnectionString);
+                simulation.IotHubConnectionStrings.Add(connString);
+            }
```

Comment:
nit: same comment here, you might even extract a method out, or update the connectionStringManager to accept in input a list